src/core/manager.c

```diff
@@ -3079,6 +3086,12 @@ void manager_check_finished(Manager *m) {
         if (m->exit_code != MANAGER_OK)
                 return;
 
+        /* For user managers, send out READY=1 as soon as we reach basic.target */
+        if (MANAGER_IS_USER(m) &&
+            unit && streq(unit->id, SPECIAL_BASIC_TARGET) &&
+            job_type == JOB_START && job_result == JOB_DONE)
+                manager_notify_ready(m, unit);
```

Comment:
hmm, i think it would be nicer to to make this "level triggered" rather than "edge triggered". that way we the function wouldn't have to have the additional arguments. i.e. something like:

```c
unit = manager_get_unit(m, SPECIAL_BASIC_TARGET);
if (!unit || !unit->job)
        manager_notify_ready(m);
```

and then manager_notify_ready() should probably maintain an internal bool in the Manager object, that sends the READY=1 only once?

Reply:
Updated.